wfc3tools/pstat.py

```diff
@@ -98,25 +164,41 @@ def pstat(filename, extname="sci", units="counts", stat="midpt", title=None,
             yend = myfile[1].header["NAXIS2"]  # full y size
 
         for i in range(1, nsamp, 1):
+            data = myfile[extname.upper(), i].data[xstart:xend, ystart:yend]
+
+            # mask the data and remove outlyier values
+            if mask is not None:
+                data = data[~mask[xstart:xend, ystart:yend]]
+
+            if high_limit is not None:
+                data = data[data < high_limit]
+
+            if low_limit is not None:
+                data = data[data > low_limit]
+
+            if data.size == 0:
+                print("No valid pixels in ext {} of {}".format(i, imagename))
+                return xaxis, yaxis
+
             if "midpt" in stat:
```

Comment:
I guess it won't make a lot of difference to the code at this point in time, but I feel like these would be more clear as an `elif` run for all the stat stuff. Since you're editing this file anyway it seems worth doing.